paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/lookup/FileStoreLookupFunction.java

```diff
@@ -361,4 +367,8 @@ protected Set<Integer> getRequireCachedBucketIds() {
         // TODO: Implement the method when Flink support bucket shuffle for lookup join.
         return null;
     }
+
+    protected void setCacheRowFilter(@Nullable Filter<InternalRow> cacheRowFilter) {
```

Comment:
Where this method be called?

Reply:
What business scenarios is this feature used in?

Reply:
This change can reduce the memory usage of the lookup join function if how the fact table rows are shuffled is known by the lookup join function. We can filter out rows that will not shuffle to this specific subtask.

For this feature to work, Flink will support specifying a custom data distribution of the fact table, please see the [FLIP](https://cwiki.apache.org/confluence/display/FLINK/FLIP-462+Support+Custom+Data+Distribution+for+Input+Stream+of+Lookup+Join).